implementations/rust/ockam/ockam_api/src/cli_state.rs

```diff
@@ -1238,6 +1245,97 @@ impl ProjectState {
     }
 }
 
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct TrustContextsState {
+    dir: PathBuf,
+}
+
+impl TrustContextsState {
+    fn new(cli_path: &Path) -> Result<Self> {
+        let dir = cli_path.join("trust_contexts");
+        std::fs::create_dir_all(dir.join("data"))?;
+        Ok(Self { dir })
+    }
+
+    pub fn create(&self, name: &str, config: TrustContextConfig) -> Result<TrustContextState> {
+        let path = {
+            let mut path = self.dir.clone();
+            path.push(format!("{name}.json"));
+            path
+        };
+        let contents = serde_json::to_string(&config)?;
+        std::fs::write(&path, contents)?;
+        if !self.default_path()?.exists() {
+            self.set_default(name)?;
+        }
+
+        Ok(TrustContextState { path })
+    }
+
+    pub fn get(&self, name: &str) -> Result<TrustContextState> {
+        let path = {
+            let mut path = self.dir.clone();
+            path.push(format!("{name}.json"));
+            if !path.exists() {
+                return Err(CliStateError::NotFound(format!("trust context `{name}`")));
+            }
+            path
+        };
+        Ok(TrustContextState { path })
+    }
+
+    pub fn default_path(&self) -> Result<PathBuf> {
+        Ok(
+            CliState::defaults_dir(self.dir.parent().expect("Should have parent"))?
+                .join("trust_context"),
+        )
+    }
+
+    pub fn default(&self) -> Result<TrustContextState> {
+        let path = std::fs::canonicalize(self.default_path()?)?;
+        Ok(TrustContextState { path })
+    }
+
+    pub fn set_default(&self, name: &str) -> Result<TrustContextState> {
+        let original = {
+            let mut path = self.dir.clone();
+            path.push(format!("{name}.json"));
+            if !path.exists() {
+                return Err(CliStateError::NotFound(format!("trust context `{name}`")));
+            }
+            path
+        };
+
+        let link = self.default_path()?;
+        std::os::unix::fs::symlink(original, link)?;
+        self.get(name)
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct TrustContextState {
+    pub path: PathBuf,
+}
```

Comment:
Is there any particular reason not to include the config here? I saw the same pattern in the `ProjectState` which differs from the rest of the states. I've found useful to be able to have the decoded config as part of the state, but maybe there is a reason not to do that?

Reply:
I will update this.

Regarding ProjectState I believe there was a reason at one point, however I don't believe it's applicable anymore.

Reply:
If it's not necessary for this PR we can add that on a separate PR. Up to you!

Reply:
No big deal to add for TC. I want to refactor this anyways, so i'm going to leave project state for now.